src/file_search.py

```
@@ -1,9 +1,37 @@
-from typing import List
+from typing import List, Optional
```

Ruff (0.14.1)

`typing.List` is deprecated, use `list` instead

(UP035)